spec/index.bs

```diff
@@ -0,0 +1,404 @@
+        text: control; url: dfn-control
+        text: extended lifetime promises; url: dfn-extend-lifetime-promises
+        text: handle functional event; url: handle-functional-event-algorithm
+        text: service worker; url: service-worker-concept
+        text: service worker registration; url: service-worker-registration-concept
+        text: termination; url: terminate-service-worker-algorithm
+    type: interface
+        text: ExtendableEvent; url: extendable-event-interface
+        text: ExtendableEventInit; url: extendable-event-init-dictionary
+        text: ServiceWorkerGlobalScope; url: service-worker-global-scope-interface
+        text: ServiceWorkerRegistration; url: service-worker-registration-interface
+
+spec: WebIDL; urlPrefix: https://heycam.github.io/webidl/#
+    type: exception
+        text: AbortError; url: aborterror
+        text: InvalidAccessError; url: invalidaccesserror
+        text: InvalidModificationError; url: invalidmodificationerror
+        text: NetworkError; url: networkerror
+        text: NotFoundError; url: notfounderror
+        text: NotSupportedError; url: notsupportederror
+        text: SecurityError; url: securityerror
+        text: SyntaxError; url: syntaxerror
+    type: interface
+        text: DOMString; url: idl-DOMString
+        text: sequence; url: idl-sequence
+</pre>
+
+<section>
+  <h2 id='introduction'>Introduction</h2>
+
+  <em>This section is non-normative.</em>
+
+  Web Applications often run in environments with unreliable networks (e.g., mobile phones) and unknown lifetimes (the browser might be killed or the user might navigate away). This makes it difficult to  synchronize client data from web apps (such as photo uploads, document changes, or composed emails) with servers. If the browser closes or the user navigates away before synchronization can complete, the app must wait until the user revisits the page to try again. This specification provides a new onsync <a>service worker</a> event which can fire <a>in the background</a> so that synchronization attempts can continue despite adverse conditions when initially requested. This API is intended to reduce the time between content creation and content synchronization with the server.
+
+  As this API relies on service workers, functionality provided by this API is only available in a <a>secure context</a>.
+</section>
+
+<section>
+  <h2 id="concepts">Concepts</h2>
+
+  The sync event is considered to run <dfn>in the background</dfn> if the user agent is either closed or no service worker clients (controlled or uncontrolled) exist for the corresponding service worker registration.
+</section>
+
+<section>
+  <h2 id="constructs">Constructs</h2>
+  A <a>service worker registration</a> has an associated <dfn>list of sync registrations</dfn> whose element type is a <a>sync registration</a>.
+
+  A <dfn>sync registration</dfn> is a tuple consisting of a <a>tag</a> and a <a lt="registration state">state</a>.
```

Comment:
My goal was to define "sync registration"  as a SyncRegistration (which already has a tag in its options) plus state and id. Then you could search by id or tag, and return the SyncRegistration directly.